oioioi/mp/controllers.py

```diff
@@ -148,17 +149,37 @@ def update_user_result_for_problem(self, result):
             if not best_submission or (score and best_submission_score < score):
                 best_submission = submission
                 best_submission_score = score
+        return best_submission
```

Comment:
consider returning a tuple of `(best_submission, best_submission_score)` to avoid a duplicated redundant DB call later in `update_user_result_for_problem`

Reply:
if we want the function signature of `get_last_scored_submission` to be common among other implementations of it, perhaps dynamically setting an "evaulated score" on the object could do the trick too
e.g. `setattr(best_submission, 'best_submission_score', best_submission_score)`
or, pass the found `ssm` within update_user_result_for_problem to get_last_scored_submission via dynamic kwargs?
If this has a negligible performance difference, keep as is.